lib/src/linting/rule.ts

```diff
@@ -9,5 +9,5 @@ export type LintingRule = (contract: ContractNode) => LintingRuleViolation[];
 
 export interface LintingRuleViolation {
   message: string;
-  source: Locatable<unknown>;
+  source?: Locatable<unknown>;
```

Comment:
What is `Locatable`? I think I asked this before and all I got was it is a table to enable access to things from different places.

Reply:
It's a wrapper around a value that also contains a location (source file and line) so that we can show helpful error messages including the location where the bad code was found. Does that make sense?